qiskit/qobj/converters/pulse_instruction.py

```diff
@@ -217,30 +217,121 @@ def convert_acquire(self, shift, instruction):
-            register_slot (list(int)): A list of register slot addresses to store results.
+            instructions_ (List[Acquire]): List of acquire instructions to bundle.
         Returns:
             dict: Dictionary of required parameters.
+        Raises:
+            PulseError: If ``instructions`` is empty.
         """
-        res = self.convert_acquire(shift, instruction)
-        if qubits:
-            res.qubits = qubits
-        if memory_slot:
-            res.memory_slot = memory_slot
-        if register_slot:
-            res.register_slot = register_slot
-        return res
+        meas_level = self._run_config.get('meas_level', 2)
+
+        t0 = None
+        duration = None
+        memory_slots = []
+        register_slots = []
+        qubits = []
+        discriminators = []
+        kernels = []
+
+        if not instructions_:
+            raise PulseError('"instructions" may not be empty.')
+
+        for instruction in instructions_:
+            qubits.append(instruction.channel.index)
+
+            instr_t0 = shift + instruction.start_time
+            if t0 is None:
+                t0 = instr_t0
+            elif t0 != instr_t0:
+                raise PulseError(
+                    'Bundled acquire instructions may not have different starting times'
+                )
+
+            instr_duration = instruction.duration
+            if duration is None:
+                duration = instr_duration
+            elif duration != instr_duration:
+                raise PulseError(
+                    'Bundled acquire instructions may not have different starting times'
+                )
+
+            if instruction.mem_slot:
+                memory_slots.append(instruction.mem_slot.index)
+
+            if meas_level == MeasLevel.CLASSIFIED:
+                # setup discriminators
+                if instruction.discriminator:
+                    discriminators.append(
+                        QobjMeasurementOption(
+                            name=instruction.discriminator.name,
+                            params=instruction.discriminator.params,
+                        ),
+                    )
+                # setup register_slots
+                if instruction.reg_slot:
+                    register_slots.append(instruction.reg_slot.index)
+
+            if meas_level in [MeasLevel.KERNELED, MeasLevel.CLASSIFIED]:
+                # setup kernels
+                if instruction.kernel:
+                    kernels.append(
+                        QobjMeasurementOption(
+                            name=instruction.kernel.name,
+                            params=instruction.kernel.params,
+                        ),
+                    )
+        command_dict = {
+            'name': 'acquire',
+            't0': t0,
+            'duration': duration,
+            'qubits': qubits,
+        }
+        if memory_slots:
```

Comment:
Can this be `[]` or `None`?

Reply:
I'm sorry I don't follow.

Reply:
You fixed it. You'd had if memory slots = None or [], then error. The else statement is gone now